### PR TITLE
Replace "q" calls in NotificationManager, UserImport & ForumManager

### PR DESCRIPTION
**Pull request by nupplaphil**
I replaced some `q` calls.

The notifications on my local node appears normal, setting seen & get all are working too.

### CODE REVIEW

File: src/Core/NotificationsManager.php [hunk 1]
```diff
@@ -55,47 +55,25 @@ private function _set_extra($notes)
...
+			$params['limit'] = $limit;
 		}
-		$r = q(
-			"SELECT * FROM `notify` WHERE `uid` = %d $filter_sql ORDER BY $order_sql $limit",
-			intval(local_user())
-		);
+
+		$dbFilter = array_merge($filter, ['uid' => local_user()]);
+
+		$r = DBA::select('notify', [], $dbFilter, $order, $params);
```
Comment (annando) [inline src/Core/NotificationsManager.php]: ```select``` returns a different array structure than ```q```.

File: src/Content/ForumManager.php [hunk 2]
```diff
@@ -196,7 +196,7 @@ public static function profileAdvanced($uid)
 	 */
 	public static function countUnseenItems()
 	{
-		$r = q(
+		$r = DBA::p(
```
Comment (MrPetovan) [inline src/Content/ForumManager.php]: Code lisibility: Please rename `$r` to `$contacts_stmt`.

File: src/Core/NotificationsManager.php (hunk 1 above)
Comment (MrPetovan) [inline src/Core/NotificationsManager.php]: Code lisibility: Please rename `$r` to `$notifys_stmt`.

File: src/Core/NotificationsManager.php [hunk 3]
```diff
@@ -113,11 +91,7 @@ public function getAll($filter = [], $order = "-date", $limit = "")
 	 */
 	public function getByID($id)
 	{
-		$r = q(
-			"SELECT * FROM `notify` WHERE `id` = %d AND `uid` = %d LIMIT 1",
-			intval($id),
-			intval(local_user())
-		);
+		$r = DBA::selectFirst('notify', ['id' => $id, 'uid' => local_user()]);
```
Comment (MrPetovan) [inline src/Core/NotificationsManager.php]: Code lisibility: Please rename `$r` to `$notifys_stmt`.

File: src/Core/NotificationsManager.php [hunk 4]
```diff
@@ -113,11 +91,7 @@ public function getAll($filter = [], $order = "-date", $limit = "")
...
 		if (DBA::isResult($r)) {
 			return $this->_set_extra($r)[0];
```
Comment (MrPetovan) [inline src/Core/NotificationsManager.php]: Are you sure this function can take a database statement as a parameter?

File: src/Core/NotificationsManager.php [hunk 5]
```diff
@@ -461,19 +424,21 @@ public function systemNotifs($seen = 0, $start = 0, $limit = 80)
 		$sql_seen = "";
 
 		if ($seen === 0) {
-			$sql_seen = " AND NOT `seen` ";
+			$filter = ['`uid` = ? AND NOT `seen`', local_user()];
+		} else {
+			$filter = ['uid' => local_user()];
```
Comment (MrPetovan) [inline src/Core/NotificationsManager.php]: Please rewrite this block like this:
```php
$filter = ['uid' => local_user()];
if ($seen === 0) {
    $filter['seen'] = false;
}
```

File: src/Core/NotificationsManager.php [hunk 6]
```diff
@@ -461,19 +424,21 @@ public function systemNotifs($seen = 0, $start = 0, $limit = 80)
...
-			"SELECT `id`, `url`, `photo`, `msg`, `date`, `seen`, `verb` FROM `notify`
-				WHERE `uid` = %d $sql_seen ORDER BY `date` DESC LIMIT %d, %d ",
-			intval(local_user()),
-			intval($start),
-			intval($limit)
-		);
+		$params = [];
+		$params['limit'] = [$start, $limit];
+
+		$r = DBA::select('notify',
```
Comment (MrPetovan) [inline src/Core/NotificationsManager.php]: Code lisibility: Please rename `$r` to `$notifys_stmt`.

File: src/Core/NotificationsManager.php [hunk 7]
```diff
@@ -596,7 +561,7 @@ public function introNotifs($all = false, $start = 0, $limit = 80)
 		}
 
 		/// @todo Fetch contact details by "Contact::getDetailsByUrl" instead of queries to contact, fcontact and gcontact
-		$r = q(
+		$r = DBA::p(
```
Comment (MrPetovan) [inline src/Core/NotificationsManager.php]: Code lisibility: Please rename `$r` to `$notifys_stmt`.

File: src/Core/UserImport.php [hunk 8]
```diff
@@ -35,9 +36,8 @@ private static function lastInsertId()
 	 */
 	private static function checkCols($table, &$arr)
 	{
-		$query = sprintf("SHOW COLUMNS IN `%s`", DBA::escape($table));
-		Logger::log("uimport: $query", Logger::DEBUG);
-		$r = q($query);
+		$r = DBStructure::getColumns($table);
```
Comment (MrPetovan) [inline src/Core/UserImport.php]: Code lisibility: Please rename `$r` to `$table_columns`.

File: src/Database/DBStructure.php [hunk 9]
```diff
@@ -844,4 +844,18 @@ public static function existsTable($table)
...
+	 * Returns the columns of a table
+	 *
+	 * @param string $table Table name
+	 *
+	 * @return array An array of the table columns
+	 * @throws Exception
+	 */
+	public static function getColumns($table)
+	{
+		$r = DBA::p("SHOW COLUMNS FROM `" . $table . "`");
```
Comment (MrPetovan) [inline src/Database/DBStructure.php]: Code lisibility: Please rename `$r` to `$columns_stmt`.

File: src/Core/NotificationsManager.php (hunk 5 above)
Comment (nupplaphil) [reply src/Core/NotificationsManager.php]: So `false` will be interpreted as  DB - `NOT NULL`. That was the missing link :-) thanks

**Comment by nupplaphil** [comment]
Tested again with my local node and some notifications..

File: src/Core/NotificationsManager.php (hunk 5 above)
Comment (MrPetovan) [reply src/Core/NotificationsManager.php]: No, it will be interpreted as ``NOT `field` ``

**Pull request merged**